src/hooks/core/websocket/useBetterWebsocketHook.ts

```diff
@@ -1,25 +1,27 @@
-import * as React from 'react';
 import Config from 'react-native-config';
 
 import useSimpleWebsocketHooks from './useSimpleWebsocketHooks';
 import TokenStorage, {ITokenEnum} from '../../../utils/storage/custom/tokenStorage';
 import {DEFAULT_PROFILE_PIC_PATH} from '../../../utils/constants';
-import {getAnonymousUserId} from '../../../utils/users';
+import {GetstreamWebsocket, WebsocketUserDataType} from './types.d';
+import {getAnonymousUserId, getUserId} from '../../../utils/users';
 
-const useBetterWebsocketHook = () => {
-  const generateUserDataUrlEncoded: (userId: string, token: string) => string = (
-    userId: string,
-    token: string
-  ) => {
+interface IUseBetterWebsocketHook {
+  lastAnonymousMessage?: GetstreamWebsocket;
+  lastSignedMessage?: GetstreamWebsocket;
+}
+
+const useBetterWebsocketHook = (): IUseBetterWebsocketHook => {
+  const generateUserDataUrlEncoded: (user: WebsocketUserDataType) => string = (user) => {
     const userData = {
-      user_id: userId,
+      user_id: user?.userId,
       user_details: {
-        id: userId,
-        name: 'AnonymousUser',
-        image: DEFAULT_PROFILE_PIC_PATH,
+        id: user?.userId,
+        name: user?.isAnonymous ? 'AnonymousUser' : user?.name,
+        image: user?.isAnonymous ? DEFAULT_PROFILE_PIC_PATH : user?.image,
         invisible: true
       },
-      user_token: token,
+      user_token: user?.token,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of `generateUserDataUrlEncoded` function is more flexible as it now accepts a `user` object of type `WebsocketUserDataType` instead of individual parameters. This allows the function to handle both anonymous and signed-in users, which improves the modularity of the code. However, there's no null check for the `user` parameter. If `user` is `null` or `undefined`, accessing properties like `userId` or `isAnonymous` will throw an error.
```diff
-  const generateUserDataUrlEncoded: (user: WebsocketUserDataType) => string = (user) => {
+  const generateUserDataUrlEncoded: (user: WebsocketUserDataType) => string = (user) => {
+    if (!user) {
+      throw new Error('User data is not provided');
+    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->